lib/DDGC/Web/Service/ActivityFeed.pm

```diff
@@ -0,0 +1,86 @@
+package DDGC::Web::Service::ActivityFeed;
+
+# ABSTRACT: Community Activity JSON service
+
+=pod
+
+=head1 NAME
+
+DDGC::Web::Service::ActivityFeed - Log / retrieve Community Activity.
+
+=head1 DESCRIPTION
+
+This service logs any activity you care to generate and allows subscription
+to these activities via a key of your choosing.
+
+=head1 MOUNT POINT
+
+This service is expected to be mounted on '/activityfeed.json'
+
+=head1 REQUEST HANDLERS
+
+=cut
+
+use DDGC::Base::Web::Service;
+use Try::Tiny;
+use List::MoreUtils qw/ uniq /;
+
+sub pagesize { 40 }
+
+sub activity_page {
+    my ( $params ) = @_;
+    $params->{page} //= 1;
+    $params->{pagesize} //= pagesize;
+    rset('ActivityFeed')->search_rs({
+        ( $params->{filter} )
+            ? ( type => { -like => $params->{filter} =~ s/\*/%/gr } )
+            : (),
+        for_user => { '=' => [ uniq( undef, $params->{user} ) ] }
+    }, {
+        order_by => { -desc => 'me.id' },
+        rows     => $params->{pagesize},
+        page     => $params->{page},
+    });
```

Comment:
That's what the relationship I defined yesterday was for (`type` is gone, replaced by `category`, `meta*` etc), so if we have a user we can say:

```perl
$user->subscriptions->activity
```

Basically, this query is out of date.

Agree on pushing searches to subs in the rs, but I am also very lazy and like having the query plan defined at the point of contact... and this is why we have fat in all layers. Yeah, think the plans-in-rs approach is probably better for this project.

I import a few specific shortcuts in the rs base class, no reason not to pull in the full set. Thanks!